### PR TITLE
(fix): Monitoring-Icon and GSB: Adding Default State, removing min-height on monitoring-icon, fixing monitoring-icon notification cut-off on GSB

### PR DESCRIPTION
**Pull request by FMorrison87**
## Brief Description

This PR encapsulates three distinct bugs and their ensuing changes:

- `rux-monitoring-icon` not having a default state
- `rux-monitoring-icon` having a min-height that was making it difficult to vertically center them without labels
- `rux-monitoring-icon` having notifications that were cut-off in some scenarios when inside of the `rux-global-status-bar`

## JIRA Link

[ASTRO-4888](https://rocketcom.atlassian.net/browse/ASTRO-4888)
[ASTRO-4883](https://rocketcom.atlassian.net/browse/ASTRO-4883)
[ASTRO-4881](https://rocketcom.atlassian.net/browse/ASTRO-4881)

## Related Issue

**GitHub Bug Links:**
[Notifications Cut-Off in Monitoring-Icon in GSB](https://github.com/RocketCommunicationsInc/astro/issues/912)
[Monitoring -Icon Min-Height](https://github.com/RocketCommunicationsInc/astro/issues/916)
[Monitoring-Icon Has No Default State](https://github.com/RocketCommunicationsInc/astro/issues/911)

## General Notes

The fixes for these issues were all related so they are all in this PR. The min-height issue was a removal of min-height on `:host`, the default state was adding in a value for a default icon, but the notification cut-off was a little bit more complex. Due to its absolutely positioned nature the notification piece of this component was being cut-off in the GSB in scenarios where there were high amounts of notifications. This presents a two-fold problem because accounting for notifications in the 3 or 4 digits only required a space increase of a few pixels; this was done. However, in the (probably rare) event that a monitoring icon was showing thousands, millions, even billions of notifications the truncation allowed for enough characters that the order of magnitude (K,M,B) was being cut-off. My assumption is that if these numbers were to appear something would have gone wrong and the operator should know the magnitude. The fix applied was to remove `overflow: hidden` from the `header` element of the GSB. This allows the notification element to be visible in the event that huge orders of magnitude need to be shown, but has no effect when smaller numbers are shown (due to the first-tier fix of the slightly expanded width). In the case were something is overflowing the GSB it is probably better to show the operator as much of the information as possible.

## Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.
- [x] Regressions are passing and/or failures are documented
- [ ] Changes have been checked in evergreen browsers

### CODE REVIEW

**Review by micahjones13 — APPROVED** [review]
Looks good to me, but let's let @markacianfrani double check it before we merge

**Review by markacianfrani — CHANGES_REQUESTED** [review]
I think we might need to keep overflow hidden on GSB. If you were to define a smaller width (p unlikely) or viewport (more likely), elements will start to break out: 

```
<div style="display: flex; justify-content: center; width: 200px;">
    <rux-global-status-bar include-icon="true" app-state="Production" app-state-color="tag1" username="Joan Smith" app-domain="Astro" app-name="Dashboard" app-version="4.0 Alpha" menu-icon="apps">
        <rux-clock></rux-clock>
        <rux-button slot="right-side">Emergency shut off</rux-button>
    </rux-global-status-bar>
</div>
```
Screen Shot 2022-11-18 at 1 33 29 PM

This one is challenging though bc the whole component isn't responsive.

**Review by jonathantneal — APPROVED** [review]
This works. I’ve left some suggestions that I think may improve its usage.

**Comment by jonathantneal** [comment]
**Suggestion**: In `packages/web-components/src/common/functional-components/MonitoringBadge/MonitoringBadge.tsx`, change the logic to ensure the badge only displays when positive numbers are used.

```diff
-             !notifications ? 'rux-advanced-status__hidden' : ''
+             notifications! > 0 ? '' : 'rux-advanced-status__hidden'
```

File: packages/web-components/src/components/rux-monitoring-icon/rux-monitoring-icon.tsx [hunk 1]
```diff
@@ -60,6 +65,20 @@ export class RuxMonitoringIcon {
...
+            '.rux-advanced-status__badge'
+        ) as HTMLElement
+        let badgeWidth: number | undefined = notificationBadge?.clientWidth
+        if (badgeWidth <= 20) {
+            this.el.style.minWidth = '84px'
+        } else if (badgeWidth >= 21 && badgeWidth < 30) {
+            this.el.style.minWidth = '88px'
+        } else {
+            this.el.style.minWidth = '104px'
+        }
```
[the remark this answers is not in the document]
Comment (FMorrison87) [reply packages/web-components/src/components/rux-monitoring-icon/rux-monitoring-icon.tsx:79]: Committed!

**Review by markacianfrani — APPROVED** [review]
I think there's an opportunity to make this less complex by talking with design a little more, but this is a good short term solution and I dont want to block this any longer

**Pull request merged**